src/components/Dropdown/Dropdown.jsx

```diff
@@ -0,0 +1,257 @@
+import React, {useState, Fragment} from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'clsx';
+import styles from './Dropdown.scss';
+
+import {ListItem, Menu, Typography} from '~/components';
+import {Separator} from '~/components/Separator';
+import {ChevronDown} from '~/tokens/icons';
+
+export const DropdownVariants = ['default', 'ghost'];
+export const DropdownSizes = ['small', 'medium'];
+
+export const Dropdown = ({
+    data,
+    label,
+    value,
+    isDisabled,
+    maxWidth,
+    variant,
+    size,
+    icon,
+    onChange,
+    className,
+    ...props
+}) => {
+    const [isOpened, setIsOpened] = useState(false);
+    const [anchorEl, setAnchorEl] = useState(null);
+    const [minWidth, setMinWith] = useState(null);
+    const isGrouped = typeof data[0].options !== 'undefined';
+    const anchorPosition = {
+        top: 'var(--spacing-nano)',
+        left: '0px'
+    };
+
+    // ---
+    // Functions to handle events
+    // ---
+    const handleOpenMenu = e => {
+        setMinWith(`${e.currentTarget.offsetWidth}px`);
+        setAnchorEl(e.currentTarget);
+        setIsOpened(true);
+    };
+
+    const handleSelect = (e, item) => {
+        if (!item.isDisabled && item.value !== value) {
+            onChange(e, item);
+        }
+
+        if (!item.isDisabled) {
```

Comment:
I think it would be better to use the return of  onChange() to see if we should close or not the dropdown (so just a simple boolean, true if ok to close and false otherwise). If the controller decides that the selection is invalid, it could be useful to keep the drop down open. But in that case we should call onChange even if `item !=== value`

Reply:
But it’s not weird to trigger the `onChange` even if the value doesn’t change?